openwrt/files/etc/udhcpc.user

```diff
@@ -22,9 +22,11 @@ case "$1" in
     fi
 
     if [ ! -z "$hostname" ]; then
-      uci set 'system.@system[0].hostname'="$hostname"
-      uci commit
       if [ `echo "$HOSTNAME" | tr '[A-Z]' '[a-z]'` != `echo "$hostname" | tr '[A-Z]' '[a-z]'` ]; then
```

Comment:
I suppose we could combine the multiple conditional statements here, but this seems more readable than that would be.

Reply:
haha, more unreadable than it is already? 

```
if [[ ! -z "$hostname" ] && [ `echo "$HOSTNAME" | tr '[A-Z]' '[a-z]'` != `echo "$hostname" | tr '[A-Z]' '[a-z]'` ]];
```

I think this change makes sense. Originally this script would reboot if the hostname changed, but that was subsequently replaced with a service reload as you can see. I still think you want the `[ ! -z "$hostname" ]` check for cases when DHCP server doesn't supply this. I can't actually think of how this would happen, but stranger things have occurred.

@sarcasticadmin it's been years since I thought about this file. Is the understanding correct? We should only `uci set` the new hostname if it changes rather than every renew and 100% of the time that is the case we should restart the services. I'm not clear on why it's being set every time, but you might have been doing this for a reason.

Reply:
I like the compound `if` @kylerisse suggested since we should handle the case where `hostname` is not set from the dhcp server side (kea) even though its unlikely.

As for why this condition exists at all: we set the hostname to the value from the `name` field in https://github.com/socallinuxexpo/scale-network/blob/ef74408a99827005e7e36f5b2b5ab78784611f1e/facts/aps/apuse.csv#L1 It was originally identified around ~Scale 19x or so that kea would push a lower case hostname: `openwrt` while by default the hostname in our builds was `OpenWRT`. Upon commit and reboot that name didnt always take (persistent memory issues on the AP) even though it was technically equivalent minus the capitalization: https://github.com/socallinuxexpo/scale-network/pull/441 covers the justification, fix, and some additional finds. 

Later we opted to restart the services that needed the hostname update instead of the whole AP so the APs dont get in a reboot loop: https://github.com/socallinuxexpo/scale-network/pull/498/commits/2d785b30b92d25cf839456a04d1a1668c1c25332

Some of this is a little fuzz but hopefully that explains why we do this.

Reply:
Wow, thank you both so much for the context and feedback!

Reply:
> I like the compound if @kylerisse suggested

Ha! @ruebenramirez suggested this, I just wrote it out to see how unreadable it would be. Seems fine, typical Bash.